iac/compositions/cat-full/service_cat_api.tf

```diff
@@ -0,0 +1,201 @@
+locals {
+  cat_api_vcap_object = {
+    #VCAP_SERVICES={"aws-s3-bucket": [{"aws_region": "..."}], "opensearch": [{"hostname": "abc", "username": "def", "password": "ghi", "port": "1234"}]}
+    opensearch = [
+      {
+        "name"     = "aws-ccs-scale-cat-opensearch" # Naming convention matters to the code
+        "hostname" = module.search_domain.opensearch_endpoint,
+      }
+    ]
+  }
+}
+
+resource "aws_lb" "cat_api" {
+  name               = "${var.resource_name_prefixes.hyphens}-ALB-CATAPI"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups = [
+    aws_security_group.cat_api_lb.id
+  ]
+  subnets = module.vpc.subnets.public.ids
+}
+
+resource "aws_route53_record" "cat_api" {
+  name            = local.fqdns.cat_api
+  allow_overwrite = true
+  type            = "A"
+  zone_id         = var.hosted_zone.id
+  alias {
+    name                   = aws_lb.cat_api.dns_name
+    zone_id                = aws_lb.cat_api.zone_id
+    evaluate_target_health = true
+  }
+}
+
+resource "aws_lb_listener" "cat_api" {
+  certificate_arn   = module.cat_api_cert.certificate_arn
+  load_balancer_arn = aws_lb.cat_api.arn
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2016-08"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.cat_api.arn
+  }
+}
+
+resource "aws_lb_target_group" "cat_api" {
+  name            = "${var.resource_name_prefixes.hyphens}-TG-CATAPI"
+  ip_address_type = "ipv4"
+  port            = "8080"
+  protocol        = "HTTP"
+  target_type     = "ip"
+  vpc_id          = module.vpc.vpc_id
+
+  health_check {
+    matcher  = "200"
+    path     = "/health"
+    port     = "8080"
+    protocol = "HTTP"
+  }
+}
+
+module "cat_api_task" {
+  source = "../../core/resource-groups/ecs-fargate-task-definition"
+
+  aws_account_id = var.aws_account_id
+  aws_region     = var.aws_region
+
+  container_definitions = {
+    cat = {
```

Comment:
In my mind I can't decide whether this should be `api` or `http` (in DMP I fell into the habit of naming containers by their function) but I don't think it should be `cat`. 

Mainly thinking of Log Streams legibility at this point; stream name will be `container/api/abcd124123` when in fact the TASK is the API and the CONTAINER is just an http server. Semantics, I know.

Reply:
That was a leftover from my standalone work, sorry. Will correct to be `http` in line with the docker image tag naming.